integration-tests/tests/src/tests/sync/flexible.ts

```diff
@@ -193,6 +193,168 @@ describe.skipIf(environment.missingServer, "Flexible sync", function () {
+          return {
+            schema: [FlexiblePersonSchema, DogSchema],
+            sync: {
+              _sessionStopPolicy: SessionStopPolicy.Immediately,
+              flexible: true,
+              user,
+              initialSubscriptions,
+            },
+          };
+        }
+
+        describe("error", function () {
+          afterEach(function () {
+            Realm.deleteFile(this.config);
+          });
+
+          it("throws an error if no `update` function is provided", async function () {
+            this.config = getConfig(this.user, {} as any);
+
+            await expect((Realm as any).open(this.config)).to.be.rejectedWith(
+              "update must be of type 'function', got (undefined)",
+            );
+          });
+
+          it("throws an error if `update` is not a function", async function () {
+            this.config = getConfig(this.user, {
+              update: "Person",
+            } as any);
+
+            await expect((Realm as any).open(this.config)).to.be.rejectedWith(
+              "update must be of type 'function', got (Person)",
+            );
+          });
+
+          it("throws an error if `rerunOnStartup` is not a boolean", async function () {
+            this.config = getConfig(this.user, {
+              update: () => {
+                // no-op
+              },
+              rerunOnStartup: "yes please",
+            } as any);
+
+            await expect((Realm as any).open(this.config)).to.be.rejectedWith(
+              /rerunOnStartup must be of type 'boolean', got.*/,
+            );
+          });
+        });
+
+        describe("success", function () {
+          type ExtraConfig = Partial<FlexibleSyncConfiguration["initialSubscriptions"]>;
+
+          function getSuccessConfig(user: Realm.User, extraConfig: ExtraConfig = {}) {
+            return getConfig(user, {
+              update: (realm) => {
+                realm.subscriptions.update((subs) => {
+                  subs.add(realm.objects(FlexiblePersonSchema.name));
+                });
+              },
+              ...extraConfig,
+            });
+          }
+
+          async function openRealm(user: Realm.User, extraConfig: ExtraConfig = {}) {
+            const config = getSuccessConfig(user, extraConfig);
+            const realm = await Realm.open(config);
+
+            return { realm, config };
+          }
+
+          async function testSuccess(
+            user: Realm.User,
+            extraConfig: Partial<FlexibleSyncConfiguration["initialSubscriptions"]> = {},
+            closeRealmAfter = true,
+          ) {
+            const { realm, config } = await openRealm(user, extraConfig);
+
+            expect(realm.subscriptions).to.have.length(1);
+            expect(realm.subscriptions.state).to.equal(Realm.App.Sync.SubscriptionsState.Complete);
+
+            if (closeRealmAfter) {
+              closeRealm(realm, config);
+              return undefined;
+            }
```

Comment:
Might make sense to wrap this in a try-finally block to ensure it happens even if the assertions above fails.